x/campaign/keeper/msg_add_shares.go

```diff
@@ -19,6 +21,17 @@ func (k msgServer) AddShares(goCtx context.Context, msg *types.MsgAddShares) (*t
 		return nil, sdkerrors.Wrapf(types.ErrCampaignNotFound, "%d", msg.CampaignID)
 	}
 
+	mainnetLaunched, err := k.IsCampaignMainnetLaunchTriggered(ctx, campaign.CampaignID)
+	if err != nil {
+		return nil, spnerrors.Critical(err.Error())
+	}
+	if mainnetLaunched {
+		return nil, sdkerrors.Wrap(types.ErrMainnetLaunchTriggered, fmt.Sprintf(
+			"mainnet %d is already launched, action prohibited",
```

Comment:
```suggestion
			"mainnet %d launch is already triggered",
```

If think `action prohibited` is not necessary, the action is prohibited since the message fails